### PR TITLE
ASU-702 API for fetching successfully mapped apartments

### PR DESCRIPTION
**Pull request by genie9**
Creates new table with mapped apartment data and api for fetching those.
Creates new Django command for deleting mapped apartments from data table for use with cronjob.

### CODE REVIEW

File: connections/tests/utils.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
+from elasticsearch_dsl import Search
+from time import sleep
+
+
+def make_apartments_sold_in_elastic(elastic_apartments: list) -> None:
+    for item in elastic_apartments:
+        if item["apartment_state_of_sale"] == "FOR_SALE":
```
Comment (tomijarvi) [inline connections/tests/utils.py:7]: Here are as well good to use Enums instead strings

**Review by tomijarvi — COMMENTED** [review]
Awesome, well-written tests. Check my comments. Keep in mind that there will be the next developer who will be looking at the code later, so the coder needs to know why things are done and what for. It is easy to know it now but when time goes on, it going to be really hard without proper commit and comment messages 😄

File: connections/tests/conftest.py [hunk 2]
```diff
@@ -27,3 +40,25 @@ def elastic_apartments():
         yield elastic_apartments
     except SkipTest:
         skip()
+
+
+@fixture()
+def broken_elastic_apartments_for_sale():
+    # oikotie broken data is in project_new_development_status
```
[the remark this answers is not in the document]
Comment (genie9) [reply connections/tests/conftest.py:47]: I am not sure what to comment, but maybe adding enums will make data more clear

[the remark this answers is not in the document]
Comment (tomijarvi) [reply connections/tests/conftest.py:47]: I'm looking for an explanation that how it is broken, what is the difference for the correct one.

[the remark this answers is not in the document]
Comment (genie9) [reply connections/tests/conftest.py]: I don't think it's necessary. In this test idea is not about what is correct data, but that data is just not valid.

**Comment by genie9** [comment]
Included updated requirements files from dependencies updating branch to see that nothing get broken.

**Review by tomijarvi — APPROVED** [review]

**Review by tomijarvi — APPROVED** [review]

File: connections/management/commands/empty_mapped_apartments_database.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+_logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    help = "Empty MappedApartment database table"
+
+    def handle(self, *args, **options):
+        try:
+            MappedApartment.objects.all().delete()
+            _logger.info("Succefully deleted mapped apartments from database")
```
Comment (jmp) [inline connections/management/commands/empty_mapped_apartments_database.py:15]: Small typo here -- should be `Successfully`.

File: connections/management/commands/empty_mapped_apartments_database.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        except Exception as e:
+            _logger.error("Could not delete mapped apartments from database:", str(e))
```
Comment (jmp) [inline connections/management/commands/empty_mapped_apartments_database.py:17]: I think a bit more Pythonic would be to use `_logger.exception("Could not delete mapped apartments from database:")`. It still uses `ERROR` log level, but will log the current exception (`e`) automatically after printing the message.

**Review by jmp — APPROVED** [review]
I think it looks fine. 👍 I added a couple of comments but they are very minor so up to you if you feel they're worth changing or not.

**Pull request merged**